### PR TITLE
Added e2e tests, refactored code

### PR DESCRIPTION
**Pull request by ieuna**

### CODE REVIEW

File: mxd/backend-service/README.md [hunk 1]
```diff
@@ -1,197 +1,147 @@
 # Tractus-X Backend Service
 
-##  API Details Summary
+## API Details Summary
 
-Backend Service is used to validate the transfer. it has the following end points.
+Backend Service is used to validate the transfer. It has the following APIs.
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:5]:
```suggestion
The Backend Service is used to simulate a data source or sink enabling the transfer of data and supporting the validation of the end-to-end process. It has the following APIs:
```

File: mxd/backend-service/README.md [hunk 2]
```diff
@@ -1,197 +1,147 @@
...
 
-## 1 Contents API
+- [Contents API](#contents-api)
+- [Transfer API](#transfer-api)
 
-### POST API /api/v1/contents to store an assets. It should return a url / id of this content in the response.
+## Contents API
 
+### Generate a random content
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:12]: Capitalisation of section titles not consistent.

File: mxd/backend-service/README.md [hunk 3]
```diff
@@ -1,197 +1,147 @@
...
-        "completed": false
-    }
-]
+{
+  "id": "3b777103-5e06-461b-90c6-1f99e597f60d",
+  "url": "http://localhost:8080/api/v1/contents/3b777103-5e06-461b-90c6-1f99e597f60d"
+}
 ```
-### GET API /api/v1/contents/random to generate some random contents.
+This URL will be used as a Endpoint in the transfer API.
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:48]:
```suggestion
This URL will be used as an endpoint in the transfer API.
```

File: mxd/backend-service/README.md [hunk 4]
```diff
@@ -1,197 +1,147 @@
...
 
-### POST API /api/v1/transfers to accept the transfer data from the connector. Connector will push data similar to this:
+## Transfer API
 
-```shell
- Body:
+### Accept the transfer data from the connector
 
-  {
+Connector will push something similar to this:
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:92]: Sounds vague.

File: mxd/backend-service/README.md [hunk 5]
```diff
@@ -1,197 +1,147 @@
...
   "authCode": "100000",
   "properties": {}
 }
 ```
 
-### GET API /api/v1/transfers/{id}/contents to return the actual assets content (i.e. the data which is stored at the above endpoint http://alice-tractusx-connector-dataplane:8081/api/public)
-
-which should return something similar to this
+### Get Transfer Content
+Get the content which is stored at the above endpoint
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:134]:
```suggestion
Get the content which is stored at the above endpoint:
```

File: mxd/backend-service/README.md [hunk 6]
```diff
@@ -1,197 +1,147 @@
...
+- Method : GET
+- URL : http://localhost/backend-service/api/v1/transfers/{id}/contents
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:136]:
```suggestion
- URL: http://localhost/backend-service/api/v1/transfers/{id}/contents
```

File: mxd/backend-service/README.md [hunk 7]
```diff
@@ -1,197 +1,147 @@
...
   "properties": {}
 }
 ```
 
-### GET API /api/v1/transfers/{id}/contents to return the actual assets content (i.e. the data which is stored at the above endpoint http://alice-tractusx-connector-dataplane:8081/api/public)
-
-which should return something similar to this
+### Get Transfer Content
+Get the content which is stored at the above endpoint
+- Method : GET
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:135]:
```suggestion
- Method: GET
```

File: mxd/backend-service/README.md [hunk 8]
```diff
@@ -1,197 +1,147 @@
...
+- URL : http://localhost/backend-service/api/v1/transfers/{id}/contents
+- Sample Response
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:137]:
```suggestion
- Sample response
```

File: mxd/backend-service/README.md [hunk 9]
```diff
@@ -1,197 +1,147 @@
...
-  "endpoint": "http://alice-tractusx-connector-dataplane:8081/api/public",
+  "endpoint": "<Content Url>",
   "authKey": "Authorization",
   "authCode": "<Auth Code>",
   "properties": {}
 }
 ```
 
-Persisting above data with transfer id as primary key.
+- Method : POST
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:103]:
```suggestion
- Method: POST
```

File: mxd/backend-service/README.md [hunk 10]
```diff
@@ -1,197 +1,147 @@
...
+- URL : http://localhost/backend-service/api/v1/transfers
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:104]:
```suggestion
- URL: http://localhost/backend-service/api/v1/transfers
```

**Review by alexandrudanciu — COMMENTED** [review]
Please revisit especially the README file.

File: mxd/backend-service/src/test/java/org/eclipse/tractusx/mxd/testfixtures/PostgresqlEndToEndInstance.java
[the remark this answers is not in the document]
Comment (ieuna) [reply mxd/backend-service/src/test/java/org/eclipse/tractusx/mxd/testfixtures/PostgresqlEndToEndInstance.java:1]: The upstream PostgresqlEndToEndInstance class has hardcoded path for sql files: [Link](https://github.com/eclipse-edc/Connector/blob/a6a17f547e923f590f782ec3629da997d0b7c080/extensions/common/sql/sql-core/src/testFixtures/java/org/eclipse/edc/sql/testfixtures/PostgresqlEndToEndInstance.java#L43)

Using upstream TestUtils class, we can read our resources folder directly: [Link](https://github.com/sap-contributions/eclipse-tractusx_tutorial-resources/blob/19573f844cf81d06b1c2950481ae7f8314e5e5eb/mxd/backend-service/src/test/java/org/eclipse/tractusx/mxd/testfixtures/PostgresqlEndToEndInstance.java#L40)

File: mxd/backend-service/src/test/java/org/eclipse/tractusx/mxd/testfixtures/PostgresRuntime.java
[the remark this answers is not in the document]
Comment (ieuna) [reply mxd/backend-service/src/test/java/org/eclipse/tractusx/mxd/testfixtures/PostgresRuntime.java:1]: The upstream Runtimes classes have hardcoded config values for the modules: [Sample](https://github.com/eclipse-edc/Connector/blob/a6a17f547e923f590f782ec3629da997d0b7c080/system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/Runtimes.java#L57)

File: mxd/backend-service/README.md [hunk 11]
```diff
@@ -1,197 +1,134 @@
 # Tractus-X Backend Service
 
-##  API Details Summary
+## API Details Summary
```
Comment (alexandrudanciu) [inline mxd/backend-service/README.md:3]: Capitalisation

**Review by alexandrudanciu — APPROVED** [review]

File: .github/dependabot.yml [hunk 12]
```diff
@@ -58,4 +58,4 @@ updates:
     open-pull-requests-limit: 5
     labels:
       - "dependencies"
-      - "npm"
+      - "npm"
```
Comment (hemantxpatel) [inline .github/dependabot.yml:61]: Remove this change, if you add a new line at the end, this change should be gone.

File: mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/controller/ContentApiController.java [hunk 13]
```diff
@@ -80,11 +70,11 @@ public String getRandomContent() {
         return this.service.getRandomContent();
     }
 
-
-    private String createJsonResponse(String id, UriInfo uriInfo) {
+    private String createJsonResponse(String id) {
         JsonNode jsonResponse = objectMapper.createObjectNode()
                 .put("id", id)
-                .put("url", UriBuilder.fromUri(uriInfo.getBaseUri())
+                .put("url", UriBuilder.fromUri("http://localhost:8080")
```
Comment (hemantxpatel) [inline mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/controller/ContentApiController.java:76]: Can we get rid of this hardcoded value.

File: mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/controller/ContentApiController.java [hunk 14]
```diff
@@ -46,18 +38,22 @@ public class ContentApiController {
 
     private final ObjectMapper objectMapper;
 
-    public ContentApiController(ContentService service, Monitor monitor, ObjectMapper objectMapper) {
+    private String host = "";
```
Comment (hemantxpatel) [inline mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/controller/ContentApiController.java:41]: Make it final without initializing.

**Pull request merged**